src/app/courses/add-course/add-course.component.ts

```diff
@@ -9,37 +13,66 @@ import { DateService } from 'src/app/services/date.service';
 })
 export class AddCourseComponent implements OnInit {
 
-  @Output() onClosePage = new EventEmitter();
-  
-  title = '';
-  description = ''
-  startDate = new Date().toLocaleDateString('ru-RU');
-  duration = null;
-  authors = '';
+  routeParams: {
+    id?: number;
+  } = {};
 
-  constructor(private courseService: CourseService, private dateService: DateService) { }
+  course = new Course();
+  private subscription: Subscription;
+  isCreate: boolean;
+  isNewOrEdit: boolean = true;
+  @Output() onOpenPage = new EventEmitter<ICourse>();
 
-  ngOnInit() {
+  startDate: string;
+
+  constructor(private route: ActivatedRoute,
+    private courseService: CourseService,
+    private dateService: DateService,
+    private breadcrumbService: BreadcrumbService,
+    private router: Router
+  ) { }
 
+  ngOnInit() {
+    this.subscription = this.route.params.subscribe((routeParams) => {
+      this.isCreate = (routeParams.id == undefined);
+      if (!this.isCreate) {
+        this.course = this.courseService.getCourse(routeParams.id);
+        this.startDate = this.dateService.formatDate(this.course.creationDate);
+        if (this.course != undefined) {
+          this.breadcrumbService.changeMessage(this.isCreate ? "New course" : this.course.title);
+          this.isNewOrEdit = true;
+        } else {
+          this.isNewOrEdit = false;
+        }
+      } else {
+        this.startDate = this.dateService.formatDate(new Date());
+      }
```

Comment:
please do your best and try to simplify it :)

also, it's a good practice to leave subscribe empty. and at the same time work with a stream by rxjs operators.
For example, here we could use `tap`

```
this.route.params.pipe(
  tap(routeParams => this.handleRouteParams(routeParams)),
).subscribe()
```